pubsub/pstest/fake.go

```diff
@@ -578,27 +586,29 @@ func (t *topic) publish(pm *pb.PubsubMessage, m *Message) {
 }
 
 type subscription struct {
-	topic      *topic
-	mu         *sync.Mutex // the server mutex, here for convenience
-	proto      *pb.Subscription
-	ackTimeout time.Duration
-	msgs       map[string]*message // unacked messages by message ID
-	streams    []*stream
-	done       chan struct{}
+	topic       *topic
+	mu          *sync.Mutex // the server mutex, here for convenience
+	proto       *pb.Subscription
+	ackTimeout  time.Duration
+	msgs        map[string]*message // unacked messages by message ID
+	streams     []*stream
+	done        chan struct{}
+	timeNowFunc func() time.Time
 }
 
-func newSubscription(t *topic, mu *sync.Mutex, ps *pb.Subscription) *subscription {
+func newSubscription(t *topic, mu *sync.Mutex, timeNowFunc func() time.Time, ps *pb.Subscription) *subscription {
 	at := time.Duration(ps.AckDeadlineSeconds) * time.Second
 	if at == 0 {
 		at = 10 * time.Second
 	}
 	return &subscription{
-		topic:      t,
-		mu:         mu,
-		proto:      ps,
-		ackTimeout: at,
-		msgs:       map[string]*message{},
-		done:       make(chan struct{}),
+		topic:       t,
+		mu:          mu,
+		proto:       ps,
+		ackTimeout:  at,
+		msgs:        map[string]*message{},
+		done:        make(chan struct{}),
+		timeNowFunc: timeNowFunc,
```

Comment:
I've tried several variants, seems to me propagating server's time func into `subscription` is the most convenience way to use fake time func